### ISSUE TITLE
CSV deployment custom labels not propagated to the Deployment resource

### ISSUE THREAD

**Issue by ppatierno** [operator-framework/operator-lifecycle-manager#2199]
I have a CSV with a deployment having the following definition:

```yaml
deployments:
        - label:
            custom-label: custom-label-value
          name: strimzi-cluster-operator.v0.22.1-5
          spec:
             ....
```

The problem is that the custom label above is not propagated to the corresponding Deployment resource which has only the following:

```yaml
labels:
    olm.deployment-spec-hash: 676b868f7f
    olm.owner: strimzi-bundle.v0.0.1-1
    olm.owner.kind: ClusterServiceVersion
    olm.owner.namespace: strimzi-operator
    operators.coreos.com/strimzi-bundle.strimzi-operator: ''
```

From the code I see that propagation should happen as it is here:

https://github.com/operator-framework/operator-lifecycle-manager/blob/15790a8a2f07fe65a3dbf5a45a54d35e20f2cce9/pkg/controller/install/deployment.go#L150

Also the test seems to pass adding custom labels:

https://github.com/operator-framework/operator-lifecycle-manager/blob/9359b9dd5d5c755a38fc3d79d64cce7bb31b81fe/pkg/controller/install/deployment_test.go#L149

Still trying to understand where is the problem or if I am making a mistake, but any feedback would be appreciated.

I have also opened PR #2198 to fix the documentation that mentions `labels` field instead of `label` which is the one defined in the code (https://github.com/operator-framework/operator-lifecycle-manager/blob/15790a8a2f07fe65a3dbf5a45a54d35e20f2cce9/vendor/github.com/operator-framework/api/pkg/operators/v1alpha1/clusterserviceversion_types.go#L68)

**Comment by ppatierno** [comment]
I have debugged and running the OLM locally (but using my remote OSD cluster) and it seems working.
I would assume that the OLM already running on my OSD cluster which doesn't work is an older version not supporting the labels feature? I noticed that the feature was added in v0.18.0 (2021-04-23) and my OSD cluster has OpenShift version 4.7.13 but I am not able to figure out the OLM version running on it.

**Comment by ppatierno** [comment]
I figured out the version:

```shell
oc exec -n openshift-operator-lifecycle-manager olm-operator-7b5bcdf788-s68tj -- olm --version
OLM version: 0.17.0
git commit: df08e59d3168b3c657c97ec45c5aebcbc9109f71
```

so applying labels is not supported in that version but starting from 0.18.0.

**Comment by ppatierno** [comment]
It seems that the feature is available starting from OpenShift 4.8. Closing this issue.
